main.qml

```diff
@@ -1,2391 +1,2386 @@
-            if(!resp.hasOwnProperty("monero") || !resp["monero"].hasOwnProperty(key)){
-                appWindow.fiatApiError("Coingecko API has error(s)");
-                return;
-            }
-            return resp["monero"][key];
-        } else if(url.startsWith("https://min-api.cryptocompare.com/data/")){
-            var key = currency === "xmreur" ? "EUR" : "USD";
-            if(!resp.hasOwnProperty(key)){
-                appWindow.fiatApiError("cryptocompare API has error(s)");
-                return;
-            }
-            return resp[key];
-        }
-    }
-
-    function fiatApiGetCurrency(url) {
-        var apis = appWindow.fiatPriceAPIs;
-        for (var api in apis){
-            if (!apis.hasOwnProperty(api))
-               continue;
-
-            for (var cur in apis[api]){
-                if(!apis[api].hasOwnProperty(cur))
-                    continue;
-
-                if (apis[api][cur] === url) {
-                    return cur;
-                }
-            }
-        }
-    }
-
-    function fiatApiJsonReceived(url, resp, error) {
-        if (error) {
-            appWindow.fiatApiError(error);
-            return;
-        }
-
-        try {
-            resp = JSON.parse(resp);
-        } catch (e) {
-            appWindow.fiatApiError("bad JSON: " + e);
-            return;
-        }
-
-        // handle incoming JSON, set ticker
-        var currency = appWindow.fiatApiGetCurrency(url);
-        if(typeof currency == "undefined"){
-            appWindow.fiatApiError("could not get currency");
-            return;
-        }
-
-        var ticker = appWindow.fiatApiParseTicker(url, resp, currency);
-        if(ticker <= 0){
-            appWindow.fiatApiError("could not get ticker");
-            return;
-        }
-
-        appWindow.fiatPrice = ticker;
-
-        appWindow.updateBalance();
-    }
-
-    function fiatApiRefresh(){
-        // trigger API call
-        if(!persistentSettings.fiatPriceEnabled)
-            return;
-
-        var userProvider = persistentSettings.fiatPriceProvider;
-        if(!appWindow.fiatPriceAPIs.hasOwnProperty(userProvider)){
-            appWindow.fiatApiError("provider \"" + userProvider + "\" not implemented");
-            return;
-        }
-
-        var provider = appWindow.fiatPriceAPIs[userProvider];
-        var userCurrency = persistentSettings.fiatPriceCurrency;
-        if(!provider.hasOwnProperty(userCurrency)){
-            appWindow.fiatApiError("currency \"" + userCurrency + "\" not implemented");
-        }
-
-        var url = provider[userCurrency];
-        network.getJSON(url, fiatApiJsonReceived);
-    }
-
-    function fiatApiCurrencySymbol() {
-        switch (persistentSettings.fiatPriceCurrency) {
-            case "xmrusd":
-                return "USD";
-            case "xmreur":
-                return "EUR";
-            default:
-                console.error("unsupported currency", persistentSettings.fiatPriceCurrency);
```

Comment:
can we have the same with this PR? if currency not in the supported list

Reply:
Why? fiatPriceCurrency will always be legit. It is set in two places when something is selected from a dropdown and those dropdowns are populated from a single source of truth.

```js
persistentSettings.fiatPriceCurrency = Object.keys(providerCurrencies)[0];
persistentSettings.fiatPriceCurrency = newCurrency;
```

Reply:
its the only thing i noticed while testing this, that i would see 'unsupported' when using release. if for whatever reason a currency is removed in a later version then that sanity check would catch it. 

here's my attempt at reducing the diffs with the older version (ignoring my deletion of monero submodule lol lets get this PR looking more like so it can be reviewed easier https://github.com/monero-project/monero-gui/compare/master...plowsof:monero-gui:fiat )

Reply:
Okay added